src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -167,6 +186,23 @@ HTTPProxy::_handleTxnBegin(shd::TxnBeginRequest&& request){
         });
 }
 
+seastar::future<std::tuple<sh::Status, shd::WriteResponse>>
+HTTPProxy::_handleWrite(K2TxnHandle& txn, shd::WriteRequest&& request, dto::SKVRecord&& k2record) {
+    return txn.write(k2record, request.isDelete, static_cast<dto::ExistencePrecondition>(request.precondition))
+        .then([](WriteResult&& result) {
+            return MakeHTTPResponse<shd::WriteResponse>(sh::Status{.code = result.status.code, .message = result.status.message}, shd::WriteResponse{});
+        });
+}
+
+seastar::future<std::tuple<sh::Status, shd::WriteResponse>>
+HTTPProxy::_handlePartialUpdate(K2TxnHandle& txn, shd::WriteRequest&& request, dto::SKVRecord&& k2record) {
+    std::vector<String> fieldsForPartialUpdate = shVectorToK2<String>(request.fieldsForPartialUpdate);
+    return txn.partialUpdate(k2record, fieldsForPartialUpdate, k2record.getKey())
```

Comment:
You can remove the third parameter here just to make it a bit clearer, partialUpdate has a default defined for it.